enhancements/ingress/add-dns-and-loadbalancer-conditions-to-managed-gateway.md

```diff
@@ -0,0 +1,621 @@
+* Set to `False` with reason `UnsupportedEndpointPublishingStrategy` when the
+publishing strategy is not LoadBalancerService
+* Set to `False` with reason `UnmanagedLoadBalancerDNS` when DNSManagementPolicy is `Unmanaged`
+* Set to `True` with reason `Normal` when DNS records should be managed by OpenShift
+
+*DNSReady Condition:*
+* Set to `Unknown` when DNSManagementPolicy is `Unmanaged` (OpenShift doesn't manage DNS, so status is unknown)
+* Set to `False` with reason `RecordNotFound` when the associated DNSRecord resource cannot be found
+* Set to `False` with reason `NoZones` when the DNSRecord has no zones configured
+* Set to `False` with reason `FailedZones` when one or more zones have failed to publish
+* Set to `False` with reason `UnknownZones` when zone status cannot be determined
+* Set to `True` with reason `Normal` when all zones have successfully published DNS records
+* Error messages include specific DNS provider errors when provisioning fails
+
+**LoadBalancer Condition Details:**
+
+*LoadBalancerReady Condition:*
+* Set to `False` with reason `ServiceNotFound` when the associated Service
+resource cannot be found
+* Set to `False` with reason `LoadBalancerPending` when the service exists but
+has no LoadBalancer ingress (external IP/hostname)
+* Set to `False` with reason `SyncLoadBalancerFailed` when cloud provider events
+indicate provisioning failures
+* Set to `True` with reason `LoadBalancerProvisioned` when the service has an
+external IP or hostname assigned
+* Error messages include cloud provider error details extracted from service events
+
+**Code Reuse:**
+* The implementation creates a shared package that contains status computation logic
+* This logic is used by both the existing IngressController status controller
+and the new Gateway status controller
+* Condition computation functions accept generic inputs (DNSRecord, Service,
+events, config) and return conditions
+* Gateway-specific wrappers convert between internal condition types
+and Gateway API `metav1.Condition` types
+
+**Resource Discovery:**
+* The controller watches the following resources to trigger reconciliation:
+  - Gateway resources in `openshift-ingress` namespace that uses GatewayClasses with 
+  `openshift.io/gateway-controller/v1` as their `.spec.controllerName` 
+  - DNSRecord resources with `gateway.networking.k8s.io/gateway-name` label
+  - Service resources with `gateway.networking.k8s.io/gateway-name` label
+  - DNS cluster configuration changes
+  - Infrastructure cluster configuration changes
+
+**Platform Detection:**
+* Platform type is determined from the cluster Infrastructure resource
+* DNS zone configuration comes from the cluster DNS resource
+
+**Condition Lifecycle:**
+* Conditions are added when a Gateway is reconciled in the `openshift-ingress` namespace
+* Conditions are updated in-place using `condutils.SetStatusCondition()` to preserve transition times
```

Comment:
Why preserve transition times?  Transition time should be updated if the condition changes, at least if it changes from true to false or vice versa.